app/models/exercise.rb

```diff
@@ -47,7 +47,6 @@ def prerequisite_exercises
 
   memoize
   def git
```

Comment:
I think we should keep this private and have a method for `update_git!` that calls `git.update!`. I feel like other things shouldn't need to access this thing's git info.

Reply:
I understand. There are quite some things that I will need to access though:

- Updating git
- Getting the HEAD commit
- Getting the current commit
- Getting the non_ignored_filepaths

Does it make sense to still do this via the `Exercise` class?